cni/network/network.go

```diff
@@ -433,10 +436,13 @@ func (plugin *netPlugin) Add(args *cniSkel.CmdArgs) error {
 		}
 	}
 
-	epDNSInfo, err := getEndpointDNSSettings(nwCfg, result, k8sNamespace)
-	if err != nil {
-		err = plugin.Errorf("Failed to getEndpointDNSSettings: %v", err)
-		return err
+	epDNSInfo := getCustomDNS(nwCfg)
+	if len(epDNSInfo.Servers) == 0 && epDNSInfo.Suffix == "" {
+		epDNSInfo, err = getEndpointDNSSettings(nwCfg, result, k8sNamespace)
```

Comment:
@saiyan86 any idea why the current implementations of getNetworkDNSSettings and getEndpointDNSSettings differ for Suffix in case of windows? Do we need to do the same for getCustomDNS?

Reply:
We had the same question here. We were also wondering if [Options](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockershim/network/cni/cni.go#L104) settings was something that was support or made sense in windows.

The DNS configuration the CNI passes also has Searches as an array where as the networking settings configuration in the cni only takes a single string.  Does that mean we only support a single search suffix?